src/secrets/aws_secretsmanager.rs

```diff
@@ -0,0 +1,111 @@
+use std::collections::{HashMap, HashSet};
+
+use aws_sdk_secretsmanager::{config, Client};
+use futures::executor;
+use vector_lib::configurable::{component::GenerateConfig, configurable_component};
+
+use crate::aws::{create_client, AwsAuthentication, ClientBuilder, RegionOrEndpoint};
+use crate::config::ProxyConfig;
+use crate::tls::TlsConfig;
+use crate::{config::SecretBackend, signal};
+
+// TODO: should this be moved to a different package, maybe aws or common?
+pub(crate) struct SecretsManagerClientBuilder;
+
+impl ClientBuilder for SecretsManagerClientBuilder {
+    type Client = Client;
+
+    fn build(config: &aws_types::SdkConfig) -> Self::Client {
+        let config = config::Builder::from(config).build();
+        Client::from_conf(config)
+    }
+}
+
+/// Configuration for the `aws_secretsmanager` secrets backend.
+#[configurable_component(secrets("aws_secretsmanager"))]
+#[derive(Clone, Debug)]
+pub struct AwsSecretsmanagerBackend {
+    /// ID of the secret to resolve.
+    pub secret_id: String,
+
+    #[serde(flatten)]
+    #[configurable(derived)]
+    pub region: RegionOrEndpoint,
+
+    #[configurable(derived)]
+    #[serde(default)]
+    pub auth: AwsAuthentication,
+
+    #[configurable(derived)]
+    pub tls: Option<TlsConfig>,
+}
+
+impl GenerateConfig for AwsSecretsmanagerBackend {
+    fn generate_config() -> toml::Value {
+        toml::Value::try_from(AwsSecretsmanagerBackend {
+            secret_id: String::from("secret-id"),
+            region: Default::default(),
+            auth: Default::default(),
+            tls: None,
+        })
+        .unwrap()
+    }
+}
+
+impl SecretBackend for AwsSecretsmanagerBackend {
+    fn retrieve(
+        &mut self,
+        secret_keys: HashSet<String>,
+        _: &mut signal::SignalRx,
+    ) -> crate::Result<HashMap<String, String>> {
+        let client: Client = executor::block_on(async {
+            return create_client::<SecretsManagerClientBuilder>(
+                &self.auth,
+                self.region.region(),
+                self.region.endpoint(),
+                &ProxyConfig::default(),
+                &self.tls,
+            )
+            .await;
+        })?;
+
+        let get_secret_value_response = executor::block_on(async {
+            return client
+                .get_secret_value()
+                .secret_id(&self.secret_id)
+                .send()
+                .await;
+        })?;
+
+        let secret_string: Result<String, String> = get_secret_value_response.secret_string.ok_or(
+            format!(
+                "secret for secret-id '{}' could not be retrieved",
+                &self.secret_id
+            )
+            .into(),
+        );
+
+        let output = serde_json::from_slice::<HashMap<String, String>>(secret_string?.as_bytes())?;
```

Comment:
```suggestion
        let output = serde_json::from_str::<HashMap<String, String>>(secret_string.as_str())?;
```

Reply:
More efficient to just use `serde_json::from_str` since we already have `String`, and `serde_json::from_slice` will have to do UTF-8 validation all over again.